City-of-Helsinki/drupal-helfi-platform-config
# UHF-9301: add email address for read_only users if missing.

# [UHF-9301](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9301)
Missing email causes problems on user's edit page if email is missing.

## What was done
Added update hook which finds all readonly users and if email is missing, create placeholder.

## How to reproduce:
- Go to elo-test or after setting up the site (before running drush updb):
  - Go to edit user with id of 10 (user/10/edit)
  - White screen should appear  

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-9301_readonly_email`
* Run `make drush-updb drush-cr`

## How to test
- Find readonly user with no email-address.
- Go to user's edit page (if using elo, you can use user with id of 10)
  - Page should load normally 



[UHF-9301]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
